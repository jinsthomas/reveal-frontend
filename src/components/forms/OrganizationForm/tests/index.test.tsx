import { cleanup, fireEvent, queryByText, render, waitFor } from '@testing-library/react';
import React from 'react';
import Teamform from '..';
import * as fixtures from '../../../../store/ducks/tests/fixtures';

afterEach(cleanup);
describe('components/forms/OrganizationForm', () => {
  it('renders without crashing', () => {
    render(<Teamform />);
  });

  it('renders team form correctly', () => {
    /** emphasizes on fields showing up  */
    const { container } = render(<TeamForm />);

    expect(container.querySelector('input[name="name"]')).toMatchSnapshot('Team name');

    expect(container.querySelector('input[id="yes"]')).toMatchSnapshot('Active');

    expect(container.querySelector('input[id="no"]')).toMatchSnapshot('Not active');

    expect(container.querySelector('button')).toMatchSnapshot('Submit Button');
    /** should always be null unless on edit view */
    expect(container.querySelector('input#identifier')).toMatchSnapshot('Null Identifier');
  });

  it('renders fields correctly in edit mode', () => {
    const props = {
      initialValues: fixtures.organization1,
    };
    const { container } = render(<TeamForm {...props} />);
    /** Team name Input should have a value in edit view */
    const teamName = container.querySelector('input[name="name"]');
    expect(teamName && teamName.getAttribute('value')).toBe('The Luang');
  });

  it('form validation works', async () => {
    const { container, getByText, getByTestId } = render(<TeamForm />);
    fireEvent.submit(getByTestId('form'));
    /** Assert Validation Response and Button disable */
    await waitFor(() => {
      expect(queryByText(container, 'Required')).not.toBeNull();
      expect(getByText('Save Team')).toBeDisabled();
    });
  });
  it('handleSubmit create a team', () => {
    // handleSubmit(() => {}, );
  });
  it('calles submission handler', async () => {
    let submitForm: () => void;
    // Here's my submitHandler mock that isn't getting called
    submitForm = jest.fn();
    const props = {
      initialValues: fixtures.organization1,
      submitForm,
    };
    const wrapper = render(<TeamForm {...props} />);
    const emailNode = wrapper.getByTestId('name') as HTMLInputElement;
    const loginButtonNode = wrapper.getByText('Save Team') as HTMLInputElement;
    // Act--------------
    // Change the input values
    fireEvent.change(emailNode, { target: { value: 'Akuko' } });
    // This should submit the form?
    fireEvent.click(loginButtonNode);
    // Assert--------------
    await waitFor(() => {
      expect(submitForm).toHaveBeenCalledTimes(1);
    });
  });
});
