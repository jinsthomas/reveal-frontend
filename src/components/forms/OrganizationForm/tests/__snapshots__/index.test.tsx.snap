// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/forms/OrganizationForm renders team form correctly: Active 1`] = `
<input
  id="yes"
  name="active"
  type="radio"
  value=""
/>
`;

exports[`components/forms/OrganizationForm renders team form correctly: Not active 1`] = `
<input
  id="no"
  name="active"
  type="radio"
  value=""
/>
`;

exports[`components/forms/OrganizationForm renders team form correctly: Null Identifier 1`] = `null`;

exports[`components/forms/OrganizationForm renders team form correctly: Submit Button 1`] = `
<button
  aria-label="Save Team"
  class="btn btn-block btn btn-primary btn btn-secondary"
  id="team-form-submit-button"
  type="submit"
>
  Save Team
</button>
`;

exports[`components/forms/OrganizationForm renders team form correctly: Team name 1`] = `
<input
  class="form-control"
  data-testid="name"
  id="name"
  name="name"
  type="text"
  value=""
/>
`;
